aiida/backends/djsite/conftest.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Configuration file for pytest tests."""
+
+from aiida.backends import BACKEND_DJANGO
+from aiida.manage.tests import get_test_backend_name
+
+if get_test_backend_name() != BACKEND_DJANGO:
```

Comment:
I notice there is still a minor issue here.

If you set `AIIDA_TEST_PROFILE`, the test manager will simply use that profile and its backend (ignoring the value of `AIIDA_TEST_BACKEND`).

I.e. when you set the `AIIDA_TEST_PROFILE` variable to an sqlalchemy test profile *without* also setting `AIIDA_TEST_BACKEND=sqlalchemy`, then the pytest test discovery will think you are running on the default django backend and discover the wrong tests.
This means, when running on an `sqlalchemy`

Note: This does not affect the CI setup, where the `AIIDA_TEST_BACKEND` variable is set correctly.

Reply:
> @greschd and me briefly looked into whether pytest offers a way to hook custom code into _before_ test discovery but we didn't find a way to do that.

If I remember correctly there _are_ ways to hook custom code into pytest, but it's just not as simple as adding a ``pytest.skip`` somewhere.

Here's the API reference for pytest hooks: https://docs.pytest.org/en/latest/reference.html#hook-reference

Either ``pytest_configure`` or ``pytest_sessionstart`` seems to be the right place for that. 

I'm guessing the "wrong backend" issue can occur any time the fixtures are used, not only for the ``aiida-core`` tests (i.e., also for plugin testing).

Reply:
> Either pytest_configure or pytest_sessionstart seems to be the right place for that code.

This looks promising. If you can find the time to look into this, that would be great.

> I'm guessing the "wrong backend" issue can occur any time the fixtures are used, not only for the aiida-core tests (i.e., also for plugin testing).

Yes, but it only affects test discovery and in practice plugins are unlikely to have backend-dependent tests.